src/middleware/authorisation.coffee

```diff
@@ -19,134 +15,57 @@ os = require 'os'
 domain = "#{os.hostname()}.#{application.name}.appMetrics"
 sdc = new SDC statsdServer
 
-matchContent = (channel, body) ->
-  if channel.matchContentRegex
-    return matchRegex channel.matchContentRegex, body
-  else if channel.matchContentXpath and channel.matchContentValue
-    return matchXpath channel.matchContentXpath, channel.matchContentValue, body
-  else if channel.matchContentJson and channel.matchContentValue
-    return matchJsonPath channel.matchContentJson, channel.matchContentValue, body
-  else if channel.matchContentXpath or channel.matchContentJson
-    # if only the match expression is given, deny access
-    # this is an invalid channel
-    logger.error 'Channel with name "' + channel.name + '" is invalid as it has a content match expression but no value to match'
-    return false
-  else
-    return true
-
-matchRegex = (regexPat, body) ->
-  regex = new RegExp regexPat
-  return regex.test body.toString()
-
-matchXpath = (xpathStr, val, xml) ->
-  doc = new dom().parseFromString(xml.toString())
-  xpathVal = xpath.select(xpathStr, doc).toString()
-  return val == xpathVal
-
-matchJsonPath = (jsonPath, val, json) ->
-  jsonObj = JSON.parse json.toString()
-  jsonVal = getJSONValByString jsonObj, jsonPath
-  return val == jsonVal.toString()
-
-# taken from http://stackoverflow.com/a/6491621/588776
-# readbility improved from the stackoverflow answer
-getJSONValByString = (jsonObj, jsonPath) ->
-  jsonPath = jsonPath.replace(/\[(\w+)\]/g, '.$1')  # convert indexes to properties
-  jsonPath = jsonPath.replace(/^\./, '')            # strip a leading dot
-  parts = jsonPath.split('.')
-  while parts.length
-    part = parts.shift()
-    if part of jsonObj
-      jsonObj = jsonObj[part]
-    else
-      return
-  return jsonObj
-
-extractContentType = (ctHeader) ->
-  index = ctHeader.indexOf ';'
-  if index isnt -1
-    return ctHeader.substring(0, index).trim()
-  else
-    return ctHeader.trim()
-
 genAuthAudit = (remoteAddress) ->
   audit = atna.nodeAuthentication remoteAddress, himSourceID, os.hostname(), atna.OUTCOME_MINOR_FAILURE
   audit = atna.wrapInSyslog audit
   return audit
 
+authoriseClient = (channel, ctx) ->
+  if ctx.authenticated? and channel.allow?
+    if ctx.authenticated.roles?
+      match = false
+      channel.allow.forEach (role) ->
+        if ((ctx.authenticated.roles.indexOf role) isnt -1)
```

Comment:
I know this was existing code, but this might be a nice opportunity to refactor this in a more coffee'ish way:
```
if role not in ctx.authenticated.roles
```